recipe/2023-11-support-insecure-registries.md

```diff
@@ -0,0 +1,185 @@
++  insecureHttp?: boolean;
+}
+```
+
+***CLI changes***
+
+Add insecure-http flag to allow insecure connection to a Bicep registry without doing an SSL check.
+#### Bicep Publish
+
+```
+rad bicep publish --file ./redis-test.bicep --target br:localhost:5000/myregistry/redis-test:v1 --insecure-http
+```
+#### Recipe Register 
+
+```
+rad recipe register cosmosdb --template-kind bicep --template-path br:localhost:5000/myregistry/redis-test:v1 --resource-type Applications.Datastores/mongoDatabases --insecure-http
+```
+
+#### Recipe List/Show
+Add a new column to recipe list/show output table.
+
+| NAME        | TYPE           | TEMPLATE KIND  | TEMPLATE VERSION | TEMPLATE | INSECURE-HTTP|
+| :--: |:--:| :--:| :--: | :--:| :--:|
+| cosmosdb      | Applications.Datastores/mongoDatabases | bicep | | localhost:5000.io/myregistry/redis-test:v1| true| 
+
+***Bicep Changes***
+
+#### Registering recipe through bicep.
+
+Add `insecureHttp` property to use insecure registry.
+```diff
+import radius as radius
+
+@description('Specifies the location for resources.')
+param location string = 'global'
+
+resource env 'Applications.Core/environments@2023-10-01-preview' = {
+  name: 'corerp-resources-recipe-env'
+  location: location
+  properties: {
+    compute: {
+      kind: 'kubernetes'
+      resourceId: 'self'
+      namespace: 'corerp-resources-recipe-env'
+    }
+    recipes: {
+      'Applications.Datastores/mongoDatabases':{
+        recipe1: {
+          templateKind: 'bicep'
+          templatePath: 'ghcr.io/testpublicrecipe/bicep/modules/mongodatabases:v1' 
+        }
+        recipe2: {
+          templateKind: 'bicep'
+          templatePath: 'localhost:5000/myregistry/mongo-test:v1' 
++          insecureHttp: true
+        }
+      }
+    }
+  }
+}
+```
+
+## Alternatives considered
+
+#### Automatically identify the insecure registry.
+
+Insecure registries are predominantly self-hosted within local environments, so registry url mostly starts with `localhost` or `127.0.0.1` that can be used to identify if `plainHttp` option need to be set on the remote repository client i.e parse the registry URL to check if it starts with `localhost` or `127.0.0.1`.
+
+This could be a quick solution involving minimal changes but it carries various downsides.
+
+- It is not supported for other insecure registries but for locally hosted ones.
+- It is not valid if locally managed registry is SSL/TLS enabled.
+- Identifying the insecure registry from our end may weaken the security.
+
+
+## Test plan
+
+#### Unit Tests
+-   Update environment conversion unit tests to validate insecureHttp property.
+-   Add cli unit tests to validate insecure-http flag.
+-   Update environment controller and config loader tests.
+
+#### Functional tests
+- 	Add a e2e test to verify insecure-http flag for `rad recipe register` and `rad bicep publish` commands
+-   Add a e2e test to deploy a recipe pushed to a locally hosted registry.
```

Comment:
Since this is specific to bicep, should we add some tests to ensure we reject the request if this flag is set for terraform recipes at registration?